### PR TITLE
Fix optimizer regression with simplifying expressions in subquery filters

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/3760

 # Rationale for this change

Fixes regression introduced in https://github.com/apache/arrow-datafusion/pull/3636 where we stopped running SimplifyExpressions a second time after converting subqueries to joins, meaning that we stopped simplifying expressions in subquery filters.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/optimizer/src/optimizer.rs [hunk 1]
```diff
@@ -144,6 +144,10 @@ impl Optimizer {
             Arc::new(DecorrelateWhereIn::new()),
             Arc::new(ScalarSubqueryToJoin::new()),
             Arc::new(SubqueryFilterToJoin::new()),
+            // simplify expressions does not simplify expressions in subqueries, so we
```
Comment (liukun4515) [inline datafusion/optimizer/src/optimizer.rs:147]: I think it's a solution to resolve this.
But We also has a way to resolve this issue by supporting the `subquery` in the rewriter of `simplify expression`.
Why not support the `subquery` in the rewriter of simplify expression?

Comment (andygrove) [reply datafusion/optimizer/src/optimizer.rs:147]: This PR just reverts the regression and adds a regression test, but I agree, would be better to handle it in `simplify_expressions`. I filed https://github.com/apache/arrow-datafusion/issues/3770 for this.

**Review by liukun4515 — APPROVED** [review]

**Comment by liukun4515** [comment]
LGTM thanks

**Comment by andygrove** [comment]
Thanks for the review @liukun4515

File: datafusion/optimizer/src/optimizer.rs (hunk 1 above)
Comment (alamb) [reply datafusion/optimizer/src/optimizer.rs:147]: I agree #3770 is a better long term solution

**Pull request merged**